lib/ansible/modules/network/eos/eos_l3_interfaces.py

```diff
@@ -0,0 +1,285 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': '<support_group>'
+}
+
+DOCUMENTATION = """
+---
+module: eos_l3_interfaces
+version_added: 2.9
+short_description: 'Manage L3 interface attributes of Arista EOS devices.'
+description: 'This module provides declarative management of Layer 3 interfaces on Arista EOS devices.'
+author: Nathaniel Case (@qalthos)
+notes:
+  - 'Tested against vEOS v4.20.x'
+options:
+  config:
+    description: A dictionary of Layer 3 interface options
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+        - Full name of the interface, i.e. Ethernet1.
+        type: str
+        required: True
+      ipv4:
+        description:
+        - List of IPv4 addresses to be set for the Layer 3 interface mentioned in I(name) option.
+        type: list
+        elements: dict
+        suboptions:
+          address:
+            description:
+            - IPv4 address to be set in the format <ipv4 address>/<mask>
+              eg. 192.0.2.1/24, or C(dhcp) to query DHCP for an IP address.
+            type: str
+          secondary:
+            description:
+            - Whether or not this address is a secondary address.
+            type: bool
+            default: False
+      ipv6:
+        description:
+        - List of IPv6 addresses to be set for the Layer 3 interface mentioned in I(name) option.
+        type: list
+        elements: dict
+        suboptions:
+          address:
+            description:
+            - IPv6 address to be set in the address format is <ipv6 address>/<mask>
+              eg. 2001:db8:2201:1::1/64 or C(auto-config) to use SLAAC to chose an address.
+            type: str
+  state:
+    description:
+    - The state the configuration should be left in
+    type: str
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    default: merged
+"""
+
+EXAMPLES = """
+# Using deleted
+
+# Before state:
+# -------------
+#
+# interface Ethernet1
+#    ip address 192.0.2.12/24
+# !
+# interface Ethernet2
+#    ipv6 address 2001:db8::1/64
+# !
+# interface Management1
+#    ip address dhcp
+#    ipv6 address auto-config
+
+- name: Delete L3 attributes of given interfaces.
+  eos_l3_interfaces:
+    config:
+      - name: Ethernet1
+      - name: Ethernet2
+    state: deleted
+
+# After state:
+# ------------
+#
+# interface Ethernet1
+# !
+# interface Ethernet2
+# !
+# interface Management1
+#    ip address dhcp
+#    ipv6 address auto-config
+
+
+# Using merged
+
+# Before state:
+# -------------
+#
+# interface Ethernet1
+#    ip address 192.0.2.12/24
+# !
+# interface Ethernet2
+#    ipv6 address 2001:db8::1/64
+# !
+# interface Management1
+#    ip address dhcp
+#    ipv6 address auto-config
+
+- name: Merge provided configuration with device configuration.
+  eos_l3_interfaces:
+    config:
+      - name: Ethernet1
+        ipv4: 198.51.100.14/24
```

Comment:
I think examples need to be updated here:
```suggestion
        ipv4: 
        - address: 198.51.100.14/24
```